### PR TITLE
Like str restriction (Fixes #1899)

### LINKED ISSUE #1899: Crash postgresql backend: TypeError: argument of type 'int' is not iterable
Apparently happens with `?like_target.version=65`

This works: https://buildhub.prod.mozaws.net/v1/buckets/build-hub/collections/releases/records?_limit=1&like_target.version=%2265%22

This crashes: https://buildhub.prod.mozaws.net/v1/buckets/build-hub/collections/releases/records?_limit=1&like_target.version=65

```
TypeError: argument of type 'int' is not iterable
(12 additional frame(s) were not displayed)
...
  File "kinto/core/resource/__init__.py", line 292, in collection_get
    include_deleted=include_deleted)
  File "kinto/core/resource/model.py", line 100, in get_records
    auth=self.auth)
  File "statsd/client.py", line 37, in _wrapped
    return_value = f(*args, **kwargs)
  File "kinto/core/storage/postgresql/__init__.py", line 634, in get_all
    modified_field)
  File "kinto/core/storage/postgresql/__init__.py", line 741, in _format_conditions
    if '*' not in value
```

A like query with a value that is not a string should return 400

Like it is done here:
https://github.com/Kinto/kinto/blob/ea5f66dd4441447292ce26a068bdb690b7de7496/kinto/core/resource/__init__.py#L1004-L1011

https://sentry.prod.mozaws.net/operations/buildhub-prod/issues/4990222/

### PR DESCRIPTION
**Pull request by fpiedrah**
Fixes #1899 

- [x] Add tests.
- [x] Add a changelog entry.
- [x] Add your name in the contributors file.

### CODE REVIEW

File: tests/core/resource/test_filter.py [hunk 1]
```diff
@@ -123,6 +123,29 @@ def test_string_filters_naively_by_value(self):
         result = self.resource.collection_get()
         self.assertEqual(len(result["data"]), 6)
 
+    def test_not_string_filter(self):
+        self.validated["querystring"] = {"like_tittle": 10}
```
Comment (Natim) [inline tests/core/resource/test_filter.py]:
```suggestion
        self.validated["querystring"] = {"like_title": 10}
```

File: tests/core/resource/test_filter.py [hunk 2]
```diff
@@ -123,6 +123,29 @@ def test_string_filters_naively_by_value(self):
...
+            error = e
+        self.assertEqual(
+            error.json,
+            {
+                "code": 400,
+                "details": [
+                    {
+                        "description": "Invalid value for like_tittle",
+                        "location": "querystring",
+                        "name": "like_tittle",
```
Comment (Natim) [inline tests/core/resource/test_filter.py]:
```suggestion
                        "name": "like_title",
```

File: kinto/core/resource/__init__.py [hunk 3]
```diff
@@ -962,6 +962,11 @@ def _extract_filters(self):
                 "description": f"Invalid value for {param}",
             }
 
+            # Return 400 if _limit is not a string
+            if "like" in param:
+                if not isinstance(value, str):
+                    raise_invalid(self.request, **error_details)
+
```
Comment (glasserc) [inline kinto/core/resource/__init__.py]: If you move this down a little bit, you can use the parsed `operator` to check whether it is exactly a `like` query (and not e.g. a check for a field like `likes_pizza`).

Comment (fpiedrah) [reply kinto/core/resource/__init__.py]: That  `like in param` could be troublesome. Fixed it now to use `operator` instead! Thanks!

**Pull request merged**